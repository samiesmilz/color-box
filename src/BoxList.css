.BoxList {
  background-color: beige;
  margin: 25px;
  padding: 10px;
  border-radius: 5px;
  overflow: hidden;
}

.BoxList-Tracker {
  font-size: small;
  background-color: azure;
  padding: 10px 10px;
  animation: growAndShrink 5s ease-in-out infinite alternate;
}

@keyframes growAndShrink {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(1.2);
  }
}
