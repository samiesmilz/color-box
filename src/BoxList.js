import React, { useState } from "react";
import NewBoxForm from "./NewBoxForm";
import "./BoxList.css";
import Box from "./Box";

const BoxList = () => {
  const [boxes, setBoxes] = useState([]); // Manage state for boxes

  const addBox = ({ width, height, color, id }) => {
    setBoxes(() => [...boxes, { width, height, color, id }]);
  };

  const removeBox = (id) => {
    setBoxes((boxes) => boxes.filter((box) => box.id !== id));
  };

  return (
    <div className="BoxList">
      <h3>Welcome to BoxList</h3>
      <NewBoxForm addBox={addBox} />
      {boxes.map(({ width, height, color, id }) => (
        <Box
          width={width}
          height={height}
          color={color}
          id={id}
          key={id}
          handleRemove={removeBox}
        />
      ))}
      <p className="BoxList-Tracker">
        Tracker: {boxes.length} boxes have been added.
      </p>
    </div>
  );
};
export default BoxList;
