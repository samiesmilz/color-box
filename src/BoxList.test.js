import { render, screen, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders withou errors", () => {
  render(<BoxList />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

const addNewBox = (width = 450, height = 250, color = "yellow") => {
  // Select input elements and the submit button
  const widthInput = screen.getByPlaceholderText("Enter width");
  const heightInput = screen.getByPlaceholderText("Enter height");
  const colorInput = screen.getByPlaceholderText("Enter color");
  const addBoxButton = screen.getByText("Add box");

  // Enter values in the input elements
  fireEvent.change(widthInput, { target: { value: width } });
  fireEvent.change(heightInput, { target: { value: height } });
  fireEvent.change(colorInput, { target: { value: color } });

  // fire an event to submit
  fireEvent.click(addBoxButton);
};

it("Adds a new box on submit", () => {
  render(<BoxList />);
  // Check to make sure there are no boxes on the page.
  expect(screen.queryByText("❇️X")).not.toBeInTheDocument();

  // Add a new box and expect to see this text now.
  addNewBox();
  expect(screen.getByText("❇️X")).toBeInTheDocument();
});

it("can remove a box", () => {
  render(<BoxList />);
  addNewBox();
  const removeButton = screen.getByText("❇️X");
  fireEvent.click(removeButton);
  expect(screen.queryByText("❇️X")).not.toBeInTheDocument();
});
