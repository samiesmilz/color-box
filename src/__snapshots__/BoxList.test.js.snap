// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <h3>
      Welcome to BoxList
    </h3>
    <div
      class="NewBoxForm"
    >
      <form>
        <label
          for="width"
        >
          Set width:
        </label>
        <input
          id="width"
          name="width"
          placeholder="Enter width"
          type="text"
          value=""
        />
        <label
          for="height"
        >
          Set height:
        </label>
        <input
          id="name"
          name="height"
          placeholder="Enter height"
          type="text"
          value=""
        />
        <label
          for="color"
        >
          Set color:
        </label>
        <input
          id="color"
          name="color"
          placeholder="Enter color"
          type="text"
          value=""
        />
        <button
          name="add-box"
          type="submit"
        >
          Add box
        </button>
      </form>
    </div>
    <p
      class="BoxList-Tracker"
    >
      Tracker: 0 boxes have been added.
    </p>
  </div>
</DocumentFragment>
`;
